README.md

```diff
@@ -1,9 +1,145 @@
+├── .eslintrc.js
+├── .prettierrc
+├── babel.config.js
+├── next.config.js
+├── package.json
+├── public
+│   ├── fonts
+│   ├── locales
+│   ├── media
+│   └── favicon.ico
+└── src
+    ├── api
+    ├── blocks
+    ├── components
+    ├── containers
+    ├── pages
+    ├── utils
+    └── pages.stories.js
+```
+
+The philosophy behind some of this structure can be read below.
+
+### src/api
+
+This is a good place to add App-level providers that should be accessible from all corners of the application. Providers added here should be exported in the api directory's common export file(s). Accessing data or helpers from these providers can look like this `import { useI18n } from 'api'`.
+
+```
+src/api
+├── __mock__ (Mock data used for Storybook)
+├── CheckoutContext (Checkout-API data & helpers)
+├── I18nContext (Internalization data & helpers)
+├── SettingsContext (CMS data)
+├── storybook.index.js
+├── index.js
+└── ...
+```
+
+### src/blocks
+
+Block components have a 1-1 relation to reusable modules created for the admin panel of chosen CMS. Ideally these should be exported from the common exports file using `next/dynamic`. This will make sure that blocks not needed on a particular page won't be downloaded to the browser unless needed.
+
+```
+src/blocks
+├── Hero
+├── Media
+├── index.js
+└── ...
+```
+
+### src/components
+
+Components found here are meant to be presentational in nature, meaning they are not supposed to be aware of any overlying structure such as page specific details or API data. Furthermore you will find the `theme` configuration here.
+
+```
+src/components
+├── colors (Color palettes)
+├── icons (Icon components)
+├── internal (MUI & OUI overrides)
+├── styles (Theme configuration)
+├── index.js
+└── ...
+```
+
+### src/containers
+
+Container components are "smart" components, they come in various sizes but what unifies them is that they have some understanding of the overlying structure. This could be page layout, page routing or API data structure. An example of this could be a `<ProductCard product={product} />` component that receives complicated data such as a product object as a prop fetched from an API.
+
+```
+src/containers
+├── App (The topmost component in the application)
+├── RouterLink (A Next.js wrapped routing component)
+├── index.js
+└── ...
+```
+
+### src/pages & src/pages.stories.js
+
+The `src/pages` directory is used by Next.js to configure page routing. [Read more about Next.js pages.](https://nextjs.org/docs/basic-features/pages)
+The `src/pages.stories.js` file is used to configure your pages for Storybook.
+
+Both Next.js & Storybook pages use the same `createRenderBlock` helper found under `src/utils/createRenderBlock.js` to iterate over Block components as described from an API. Using the same setup helps to keep a consistent codebase across both environments. Don't forget to add your newly created Block component to the common exports file!
+
+## One template, two enhancers
```

Comment:
Maybe change "enhancers" to "entry points"? The index files might not contain any enhancing logic.

Reply:
Let's change that!